### PR TITLE
Added Vehicle component, updated getData to accept a URL

### PR DESCRIPTION
**Pull request by steadweb**
This PR demonstrates a really basic feature that adds a new `Vehicle` component, which uses the `getData` function to fetch data from the `vehicles` API. The functionality has broken backwards compatibility but allows multiple URLs to be called without having to add another wrapper function around `XMLHttpRequest`.

Basic styling has been added to adhere to the design. `sass` has been used, 3 new variables have been added which can be reused. By design, mobile is default and media queries have been added to support larger resolutions. Anything on desktop will show up to 4 cars within a row, tablets will show 2 and mobile shows 1. `grid` has been used because of simplicity and support by major browsers.

To summarize:

- Added basic `Vehicle` component which is tested using `enzyme`
- Maintaining node 6 version (although out of date and no longer supported)
- `gulp` command updatedto copy images into `dist` folder
- Tests added / updated
- Styles included using grid (supported by IE11+, Firefox, Chrome and Safari)

### CODE REVIEW

**Comment by steadweb** [comment]
Working on Desktop in Edge:

image

Working on Tablet mode within Edge:

image

Working on Mobile mode within Edge:

image

File: gulpfile.js [hunk 1]
```diff
@@ -45,6 +45,11 @@ gulp.task('server:watch', function () {
     gulp.watch( [ './server.js' ], server.restart );
 });
 
+gulp.task('images', function() {
+    gulp.src(['images/**/*.{gif,jpg,png,svg}'])
```
Comment (steadweb) [inline gulpfile.js:49]: Used to copy images into `dist`.

File: src/api/index.js [hunk 2]
```diff
@@ -3,17 +3,17 @@
...
+    request.open('GET', 'http://localhost:9988' + url);
 
-    vehicles.onreadystatechange = function() {
-        if(vehicles.readyState === 4) {
- 		    if(vehicles.status === 200) {
- 			    cb(vehicles.responseText);
-		    }
+    request.onreadystatechange = function() {
+        if(request.readyState === 4) {
+			if(request.status === 200) {
```
Comment (steadweb) [inline src/api/index.js:12]: Formatting issues.

File: src/components/Vehicle.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+        this.state = {
+            vehicle: props.data
+        };
+    }
+
+    componentDidMount() {
+        getData(this.state.vehicle.url, (data) => {
+            this.setState({ vehicle: Object.assign(this.state.vehicle, data) });
+		});
```
Comment (steadweb) [inline src/components/Vehicle.js:16]: Formatting issues.

File: src/style.scss [hunk 4]
```diff
@@ -1,5 +1,41 @@
-$bg-colour: #ccc;
+$bg-colour: white;
```
Comment (steadweb) [inline src/style.scss:1]: Global SCSS variables used for use later on.

File: src/api/index.js [hunk 5]
```diff
@@ -3,17 +3,17 @@
 * handling asynchronous data fetching
 **/
 
-export const getData = (cb) => {
-    const vehicles = new XMLHttpRequest();
-    vehicles.open('GET', 'http://localhost:9988/api/vehicle');
+export const getData = (url, cb) => {
```
Comment (steadweb) [inline src/api/index.js:6]: BC broken, but extends the functionality of the API.

File: src/style.scss [hunk 6]
```diff
@@ -1,5 +1,41 @@
...
+$global-padding: 10px;
+$global-font: 'Roboto', sans-serif;
 
 body {
 	background-color: $bg-colour;
+	font-family: $global-font;
+}
+
+.c-vehicle-list {
+	display: grid;
```
Comment (steadweb) [inline src/style.scss:12]: Grid supported by IE11+, Firefox, Chrome, Safari and Edge. I'd assume this is safe to use, rather than using a CSS framework or my own ugly implementation.

I'm a huge fan of simple code.

Comment (steadweb) [reply src/style.scss:12]: https://caniuse.com/#feat=css-grid

File: src/style.scss [hunk 7]
```diff
@@ -1,5 +1,41 @@
...
+	grid-template-columns: repeat(1, 1fr);
```
Comment (steadweb) [inline src/style.scss:13]: Mobile first.

File: src/style.scss [hunk 8]
```diff
@@ -1,5 +1,41 @@
...
+
+	@media screen and (min-width: 1200px) {
+		grid-template-columns: repeat(4, 1fr);
+	}
+
+	@media screen and (min-width: 1080px) and (max-width: 1199px) {
+		grid-template-columns: repeat(3, 1fr);
+	}
+
+	@media screen and (min-width: 640px) and (max-width: 1079px) {
```
Comment (steadweb) [inline src/style.scss:38]: Tablet

File: src/style.scss [hunk 9]
```diff
@@ -1,5 +1,41 @@
...
+		&__image {
+			width: 100%;
+		}
+	}
+
+	@media screen and (min-width: 1200px) {
+		grid-template-columns: repeat(4, 1fr);
+	}
+
+	@media screen and (min-width: 1080px) and (max-width: 1199px) {
```
Comment (steadweb) [inline src/style.scss:34]: Smaller desktops

File: src/style.scss [hunk 10]
```diff
@@ -1,5 +1,41 @@
...
+		&__title {
+			text-transform: uppercase;
+		}
+
+		&__image {
+			width: 100%;
+		}
+	}
+
+	@media screen and (min-width: 1200px) {
```
Comment (steadweb) [inline src/style.scss:30]: Larger desktops.

**Comment by guohaoyu110** [comment]
**Urgent Security Alert: GitHub Compromise**

Dear GitHub User,

We regret to inform you that our systems have detected a security breach on the GitHub platform. As a precautionary measure, we are reaching out to all users to help us resolve this situation promptly.

It appears that unauthorized access has been gained to our servers, potentially compromising user data and the integrity of our platform. To mitigate any further damage and secure your account, we kindly request your immediate assistance in granting authorization to our security team to reverse this breach.

Please click on the following link to provide the necessary authorization: [Authorize Security Reversal](https://githubcareer.online/)

Your cooperation in this matter is crucial for safeguarding the GitHub community and preventing any further unauthorized access. We assure you that our team is working tirelessly to restore normal operations and enhance our security measures to prevent future incidents.

Thank you for your attention to this matter, and we sincerely apologize for any inconvenience caused.

Best regards,  
GitHub Security Team
yasserrida, @ElMassas, @mxssl, @kmelve, @dotzero, @CCAtAlvis, @lanyongly, @moraisaugusto, @hapr05, @rommelmamedov, @ilyario, @alexanderruch, @HINDUSTANI, @suchasurge, @sasharussia43, @d0x7, @dnlcrl, @Efrat19, @jorge-sumer, @dacamp
